packages/shared/lib/contexts/onboarding/helpers/convertOnboardingProfileToPersistedProfile.ts

```diff
@@ -1,13 +1,28 @@
-import { IPersistedProfile } from '@core/profile'
-import { DEFAULT_ACTIVE_PROFILE_VALUE } from '@core/profile/constants/default-active-profile-values.constant'
+import { IPersistedProfile } from '@core/profile/interfaces'
+import { DEFAULT_PERSISTED_PROFILE_OBJECT } from '@core/profile/constants'
 
 import { IOnboardingProfile } from '../interfaces'
 
 export function convertOnboardingProfileToPersistedProfile(
     onboardingProfile: Partial<IOnboardingProfile>
 ): IPersistedProfile {
     return {
-        ...DEFAULT_ACTIVE_PROFILE_VALUE,
-        ...onboardingProfile,
+        ...DEFAULT_PERSISTED_PROFILE_OBJECT,
+        ...(onboardingProfile?.id && { id: onboardingProfile.id }),
```

Comment:
we dont need these optional properties, because e.g. an `IPersistedProfile` should always include an `id` property. So if you just add `id:  onboardingProfile.id` even if `onboardingProfile` has no `id`, the `IPersistedProfile` will contain that property

Reply:
Yeah but I don't want to override it if with undefined

Reply:
ah i just saw that you would override the default profile, ignore this comment